src/codegen/compile.ml

```diff
@@ -1600,15 +1610,18 @@ module Tagged = struct
       set_object ^^ get_object ^^
       compile_unboxed_const (int_of_tag tag) ^^
       Heap.store_field tag_field ^^
-      get_object ^^ (* object pointer *)
-      get_object ^^ (* forwarding pointer *)
-      Heap.store_field forwarding_pointer_field ^^
+      (if !Flags.gc_strategy == Flags.Incremental then
```

Comment:
```suggestion
      (if !Flags.gc_strategy = Flags.Incremental then
```
use structural equality

Reply:
Thank you very much. I fixed this (including the other cases).